packages/xrpl/test/integration/transactions/permissionedDEX.test.ts

```diff
@@ -0,0 +1,359 @@
+        'DirectoryNode',
+      )
+      assert.equal(
+        (directoryNode_ledger_object.result.node as DirectoryNode).DomainID,
+        pd_ledger_object.index,
+      )
+    },
+    TIMEOUT,
+  )
+
+  it(`Validate the bookOffers method`, async () => {
+    const response: BookOffersResponse = await testContext.client.request({
+      command: 'book_offers',
+      taker: wallet2.classicAddress,
+      taker_pays: {
+        currency: 'USD',
+        issuer: testContext.wallet.classicAddress,
+      },
+      taker_gets: {
+        currency: 'XRP',
+      },
+      domain: pd_ledger_object.index,
+    } as BookOffersRequest)
+
+    assert.equal(response.result.offers.length, 1)
+    assert.equal(response.result.offers[0].TakerGets, '1000')
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).value,
+      '10',
+    )
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).currency,
+      'USD',
+    )
+    assert.equal(
+      (response.result.offers[0].TakerPays as IssuedCurrencyAmount).issuer,
+      testContext.wallet.classicAddress,
+    )
+
+    assert.equal(response.result.offers[0].DomainID, pd_ledger_object.index)
+  })
+
+  it(
+    'Validate the ripple_path_find response',
+    async () => {
+      // wallet2 requests a path to wallet1, through the USD Token
+      const ripplePathFind: RipplePathFindRequest = {
+        command: 'ripple_path_find',
+        source_account: wallet2.classicAddress,
+        destination_account: wallet1.classicAddress,
+        destination_amount: {
+          currency: 'USD',
+          issuer: testContext.wallet.classicAddress,
+          value: '10',
+        },
+        source_currencies: [
+          {
+            currency: 'USD',
+            issuer: testContext.wallet.classicAddress,
+          },
+        ],
+        domain: pd_ledger_object.index,
+      }
+
+      const response: RipplePathFindResponse = await testContext.client.request(
+        ripplePathFind,
+      )
+
+      console.log(response)
+
+      assert.equal(response.result.destination_account, wallet1.classicAddress)
+      assert.equal(response.result.source_account, wallet2.classicAddress)
+      assert.equal(response.result.destination_currencies, ['USD'])
+      assert.isNotEmpty(response.result.alternatives)
+      assert.equal(response.result.destination_amount, '10')
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'Crossing a PermissionedDEX Offer',
+    async () => {
+      // wallet2 "crosses" the offer within the domain
+      const offerCrossTx: OfferCreate = {
+        TransactionType: 'OfferCreate',
+        Account: wallet2.classicAddress,
+        TakerPays: '1000',
+        TakerGets: {
+          currency: 'USD',
+          issuer: testContext.wallet.classicAddress,
+          value: '10',
+        },
+        DomainID: pd_ledger_object.index,
+      }
+
+      await testTransaction(testContext.client, offerCrossTx, wallet2)
```

Comment:
@shawnxie999  Hello Shawn, can you help me with this test case? I expect this Offer to cross with another Offer that I created in the `beforeAll` method, at the beginning of this file.

However, these offers do not cross with each other. I have ensured that they both use the same `DomainID`. The `TakerPays` and `TakerGets` values have been configured to ensure that both the offers are complementary to each other.

Do you know why they do not cross? Am I missing something?

Reply:
not sure if it will fix the issue, but can you trying making the first offer non-hybrid?

Reply:
I tried your suggestion -- it did not help ://

Reply:
you may have to enable defaultRipple on ur account root. that might be the issue